File: src/business-rules/customers/handlers/getall-customer.handler.ts
import { IGetAllCustomerHandler } from './igetall-customer.handler';
import { ICustomerListResponse } from '../responses';
import { Inject, Injectable } from '@nestjs/common';
import {
  ICustomerRepository,
  ICustomerRepositoryName,
} from '@domain/repositories';

@Injectable()
export class GetAllCustomerHandler implements IGetAllCustomerHandler {
  constructor(
    @Inject(ICustomerRepositoryName) private repository: ICustomerRepository,
  ) {}

  async execute(): Promise<ICustomerListResponse> {
    const customers = await this.repository.findAll();

    const response: ICustomerListResponse = {
      payload: customers.map((c) => ({
        id: c.id,
        name: c.name,
        email: c.email,
        birthDate: c.birthDate,
      })),
    };

    return response;
  }
}

File: dotnet/BusinessRules/Customers/Handlers/GetAllCustomerHandler.cs
using System.Linq;
using dotnet.BusinessRules.Customers.Responses;
using dotnet.Domain.Repositories;

namespace dotnet.BusinessRules.Customers.Handlers
{
    public class GetAllCustomerHandler : IGetAllCustomerHandler
    {
        private readonly ICustomerRepository _customerRepository;

        public GetAllCustomerHandler(
            ICustomerRepository customerRepository
        )
        {
            _customerRepository = customerRepository;
        }

        public CustomerListResponse ExecuteAsync()
        {
            var customers = _customerRepository.FindAll();
            return new CustomerListResponse
            {
                Payload = customers
                    .Select(customer => new CustomerItemResponse
                    {
                        Id = customer.Id,
                        Name = customer.Name,
                        Email = customer.Email,
                        BirthDate = customer.BirthDate
                    })
                    .ToList()
            };
        }
    }
}